packages/aws-cdk-lib/aws-s3-deployment/README.md

```diff
@@ -199,6 +199,41 @@ new s3deploy.BucketDeployment(this, 'DeployMeWithoutDeletingFilesOnDestination',
 });
 ```
 
+### Explicit Prune Validation
+
+To prevent accidental data loss, you can enable the feature flag `@aws-cdk/aws-s3-deployment:requireExplicitPrune` 
+in your `cdk.json` to require explicit setting of the `prune` property:
+
+```json
+{
+  "context": {
+    "@aws-cdk/aws-s3-deployment:requireExplicitPrune": true
+  }
+}
+```
+
+When this feature flag is enabled, you must explicitly set `prune` to `true` or `false`:
+
+```ts
+declare const destinationBucket: s3.Bucket;
+new s3deploy.BucketDeployment(this, 'MyDeployment', {
+  sources: [s3deploy.Source.asset(path.join(__dirname, 'my-website'))],
+  destinationBucket,
+  prune: true, // Must be explicitly set when feature flag is enabled
```

Comment:
should also update the `prune` documentation to make it clear that we recommend explicitly setting this property